### DISCUSSION TITLE
Planificación término del soporte del plugin

### DISCUSSION THREAD

**Discussion by EstebanForge** [whooohq/whq-woocommerce-chilexpress-shipping#246] [General]
Ver: https://github.com/whooohq/whq-woocommerce-chilexpress-shipping/issues/245#issuecomment-1194210887

Se debe planificar una salida responsable para el plugin. ¿En qué sentido?

1. Se debe informar a los usuarios actuales del problema. ¿Dentro del plugin? ¿En qué partes de WP?
2. Se debe informar en el repo oficial del plugin en WP, update del readme.
3. Se debe informar en el repo oficial acá en GitHub, update del readme.
4. ¿Por cuánto tiempo se debe mantener el plugin arriba en los repos de WP, antes de pedir la baja del mismo? ¿Cuánto tiempo es prudente para que el aviso lo vean los usuarios del plugin? ¿Un mes? ¿Meses?.
5. ¿Qué alternativas se entregan?, aparte de la obvia, plugin oficial de Chilexpress. ¿Starken? (tiene plugin oficial). ¿Otras de terceros con integración a WooCommerce andando, como: enviame.io, envia.com? ¿otras con plugin o integración WooCommerce andando, cuáles?
6. Agradecer a todos los que confiaron en este (complicado*) desarrollo a lo largo de los años que pudo sobrevivir. (*) Por no tener documentación oficial de Chilexpress de la API WSDL, nunca. Por no haber recibido ayuda de ellos, a pesar de haberla solicitado formalmente algunas veces, y mucho antes que existiese la API REST que ahora promocionan (y venden). Esta larga y triste historia, por si alguien la quiere conocer, ha quedado documentada por ahí entre todos los issues que se levantaron para este plugin. Fue una larga historia, desde junio del 2017 a la fecha.
8. Definir fecha de cierre final.

Eso, básicamente.

Esteban

**Comment by PatrickCaneloDigital** [comment]
1)
Sugiero dentro del plugin, no para el publico sino en el admin-board. Adicionalmente en el REPO WP y GITHUB desde ya.

Con este hook, se pueden generar mensajes de advertencia en el backend:


`
function wpb_admin_notice_warn() {
echo '<div class="notice notice-warning is-dismissible">
      <p>Importante: Lamentamos informar que desde la fecha ... el plugin woocommerce chilexpress shipping quedara sin soporte.<p>Sentimos tener que dar este paso, pero loc cambios de chilexpress migrando a una nueva API Rest en donde no se tiene acceso a producción sin pagar, nos hace muy dificil seguir manteniendolo. Durante varios años hemos invertido nuestro tiempo libre en ir haciendo ajustes y adaptaciones, pero llegamos al punto que no podemos entregar mas soporte con nuestro tiempo libre.</p>
      <p>Desde la fecha ... el plugin quedará sin soporte y a partir de ... será eliminado del repositorio wordpress</p>
      <p>alternativas serán: ....</p>
      </div>'; }
add_action( 'admin_notices', 'wpb_admin_notice_warn' );
`

sugiero no hacerlo "dismissible" > es decir eliminar esta clase y eventualmente no un warning sino en rojo: notice-error
existen estas clases:
    notice-warning
    notice-error
    notice-info
    notice-success
    
1 b) igual se podría incluir en la actualización un correo unico (que se envia durante la activacion del plugin) al admin, advirtiendo de la situación

4) Meses creo... tipo fin de año. Ya que no se sabe bien cuantas descargas del plugin están siendo ocupadas (sospecho que muchas), hay que ser prudente con los tiempos, porque no creo que muchos se den cuenta al tiro, aunque con el mail al admin quizas si... es decir, descontinuar el soporte puede ser apartir de ahora, pero la eliminación del repositorio, hay que dar tiempo que la gente logre actualizar a la versión que les advierte del "takedown" y de allí darles otro tiempo para buscar e implementar alternativas... (porque puede tener hasta implicancias en las tiendas, nuevos contratos con servicios shipping si es que)

5) yo mencionaría como "alternativa" adicionalmente a las mencionadas, el mismo github, tipo plugin sin soporte y explicando como ocupar la API Rest, explicando problemas de trabajar solo con llaves dev y como pueden sonseguir las llaves PROD. Porque hasta donde yo se, este plugin es la unica alternativa real al plugin oficial de chilexpress. Quizas me equivoco

**Reply by nicolasfranz** [reply]
@TCattd un último agradecimiento por este desarrollo, así como a toda la comunidad que participó en generar mejoras incrementales. Una pena que C.Express haya tomado el camino actual, pero bien, hay que seguir avanzando. Respecto a tu invitación a reflexionar, algunos alcances en donde me gustaría contribuír:

Punto 1: Se debe informar a los usuarios actuales del problema. ¿Dentro del plugin? ¿En qué partes de WP?

La sugerencia de Patrick es justo lo que estaría comentando aquí. Una notificación en la interfaz, de forma persistente, entrega lo mínimo necesario para que los shop managers y administradores de cada sitio, puedan tomar las medidas necesarias.

Puntos 2 y 3, nada que agregar, es necesario. Importante ser muy abierto al respecto transmitiendo que esta es una decisión que se ha tomado a partir de las acciones de C.Express con respecto al plugin gratuito, y la nueva dinámica de activación de tal integración, que requiere pasar por el uso del plugin oficial de ellos, etc. etc. etc.

Punto 4: Estimo que 6 meses es más que suficiente, ahora bien, hay casos de casos. Personas que no revisan con frecuencia su sitio, y el gran tema aquí es que algunas empresas no tienen políticas de mantención frecuentes. Siendo así, solo enterarán cuando vean caídas en sus conversiones finales. Tal vez sea más prudente cortar a los 2/3 meses máximo posterior del aviso, para evitarles a tales empresas/profesionales que se extienda su falta de conversiones a razón simple de no estar monitoreando la experiencia de sus usuarios en la web.

Punto 5: los agradecimientos aquí se repiten para todos, y en especial para ti Esteban por toda esta contribución a la comunidad. Al respecto de alternativas! Pocas conozco, pero estamos armando algo en específico para Enviame, lo enviaré para iniciar una discusión al respecto.

**Comment by EstebanForge** [comment]
Agregado un notice en el plugin. Se puede cerrar, pero volverá, todas las semanas. Existe un evento programado corriendo, que limpia el checkeo todas las semanas, para que no digan que no vieron el aviso.
El aviso es breve y claro: https://app.box.com/s/e67ncil49ce8v7mnxwm94ysqctfg6ekq
Y lleva acá: https://github.com/whooohq/whq-woocommerce-chilexpress-shipping/discussions/248

Lo del correo no es mala idea. De hacerlo, tendría que probarlo bien, para no dejar la escoba si algo sale mal.
Podría hacer un checkeo como el del notice, escribir una opción una vez enviado el email, para que salga una sola vez, y solo al admin del sitio (y/o tienda). Un solo correo, una sola vez.

Fecha oficial de cierre: 1ero de Enero del 2023.

Se agregó aviso en el repositorio acá, y en la página del plugin en wordpress.org

Llegada la fecha de cierre, se pedirá al equipo de wordpress.org que bajen el plugin.

Esteban
